### PR TITLE
Ithc/dsi 6937 ithc request headers changes

### PR DESCRIPTION
**Pull request by malikimran34**
Jira ticket
https://dfe-secureaccess.atlassian.net.mcas.ms/jira/software/c/projects/DSI/boards/14?selectedIssue=DSI-6937

Content security policy has been refactored to exclude localhsot for live environments

- Add the Header X-Contact-Type-Options
- Set the Strict-Transport-Security where headers missing
- Remove CSP headers extra information
- Set X-Frame-Options
- Set X-XSS-Protection
- Set X-Content-Type-Options
- Remove fingerprints of server e.g. powered by express

Progress https://dfe-secureaccess.atlassian.net.mcas.ms/wiki/spaces/NSA/pages/3947528194/ITHC+April+2024+-+Medium+Priority+Recommendations

### CODE REVIEW

File: package.json
Comment (trickyward) [inline package.json:1]: Only a minor point, but I wonder whether we can reference the latest DfE repos, for example, async-retry: 2.0.3, dao: 4.2.13, auth: 2.3.1.

**Review by trickyward — COMMENTED** [review]
Only a minor point about referencing the latest DfE repos.

File: package.json
Comment (malikimran34) [reply package.json:1]: let me check if there are not containing major changes i will upgrade them. thanks

Comment (malikimran34) [reply package.json:1]: that's all done. all packages are up to date. thanks

**Review by trickyward — APPROVED** [review]
Looks good to me, and consistent with other implementations.

**Review by kruncher — APPROVED** [review]

**Pull request merged**